addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/meter/DSMRMeter.java

```diff
@@ -0,0 +1,160 @@
+ * - {@link DSMRMeterKind} describing what kind of meter (electricity, gas, etc.) this is. This information
+ * is mainly needed for the autodiscovery proces since a set of {@link CosemObject} could fit multiple meter types
+ * (i.e. Electricity for DSMR can have meter type ELECTRICITY_V4 and ELECTRICITY_V404. The meter kind for both types
+ * is ELECTRICITY. The auto discovery proces knows know both types describe the same kind and can find the most
+ * appropriate one.
+ * - {@link DSMRMeterIdentification} describing the identification of this meter.
+ *
+ * {@link DSMRMeterType} and {@link DSMRMeterIdentification} are private members of the DSMRMeter
+ * If both 3 members are equal both meters are the referring the same physical meter part of the DSMR
+ * physical device
+ *
+ * @author M. Volaart - Initial contribution
+ */
+public class DSMRMeter {
+    private final Logger logger = LoggerFactory.getLogger(DSMRMeter.class);
+
+    /**
+     * Meter identification
+     */
+    private final DSMRMeterDescriptor meterDescriptor;
+
+    /**
+     * List of supported message identifiers for this meter
+     */
+    private List<OBISIdentifier> supportedIdentifiers;
+
+    /**
+     * Listener of new meter values
+     */
+    private DSMRMeterListener meterListener;
+
+    /**
+     * Creates a new DSMRMeter
+     *
+     * @param meterDescriptor {@link DSMRMeterDescriptor} containing the description of the new meter
+     * @param meterListener {@link DSMRMeterListener} containing the listener for new Cosem Objects
+     */
+    public DSMRMeter(DSMRMeterDescriptor meterDescriptor, DSMRMeterListener meterListener) {
+        if (meterListener == null) {
+            throw new IllegalArgumentException("meterListener can not be null");
+        }
+        this.meterDescriptor = meterDescriptor;
+        this.meterListener = meterListener;
+
+        supportedIdentifiers = new LinkedList<>();
+
+        for (CosemObjectType msgType : meterDescriptor.getMeterType().supportedCosemObjects) {
+            OBISIdentifier obisId = msgType.obisId;
+            if (msgType.obisId.getGroupB() == null) {
+                supportedIdentifiers.add(new OBISIdentifier(obisId.getGroupA(), meterDescriptor.getChannel(),
+                        obisId.getGroupC(), obisId.getGroupD(), obisId.getGroupE(), msgType.obisId.getGroupF()));
+            } else {
+                supportedIdentifiers.add(msgType.obisId);
+            }
+        }
+    }
+
+    /**
+     * Returns a list of Cosem Objects this meter will handle
+     *
+     * @param cosemObjects the list of CosemObject that must be processed
+     * @return List of CosemObject that this meter can process
+     */
+    private List<CosemObject> filterMeterValues(List<CosemObject> cosemObjects) {
+        List<CosemObject> filteredList;
+
+        logger.trace("supported identifiers: {}, searching for objects {}", supportedIdentifiers, cosemObjects);
+        filteredList = cosemObjects.stream()
```

Comment:
This can be simplified to simply return here. The variable `filteredList` isn't really used.